tests/storage/test_event_federation.py

```diff
@@ -506,11 +508,24 @@ def insert_event(txn):
         )
         self.assertSetEqual(difference, set())
 
-    def test_prune_inbound_federation_queue(self):
-        "Test that pruning of inbound federation queues work"
+    @parameterized.expand(
+        [
+            (RoomVersions.V1,),
+            (RoomVersions.V9,),
+        ]
+    )
+    def test_prune_inbound_federation_queue(self, room_version: RoomVersion):
+        """Test that pruning of inbound federation queues work"""
 
         room_id = "some_room_id"
 
+        def prev_event_format(prev_event_id: str) -> Union[Tuple[str, dict], str]:
+            """Account for differences in prev_events format across room versions"""
+            if room_version == RoomVersions.V1:
```

Comment:
If we change the test parameters to include all known room versions, this condition needs to also include v2 rooms according to the spec